web/libs/editor/src/regions/Result.js

```diff
@@ -158,6 +158,24 @@ const Result = types
     get canBeSubmitted() {
       const control = self.from_name;
 
+      // Find the first node moving up in the tree with the given visibleWhen value
+      function findNodeWithVisibleWhen(control, visibleWhen) {
+        let currentControl = control;
+
+        while (currentControl) {
+          if (currentControl.visiblewhen === visibleWhen) return currentControl;
+
+          try {
+            currentControl = getParent(currentControl);
+            if (!currentControl) break;
+          } catch {
```

Comment:
do we expect it to break? I checked our usage of this function and there are no try-catch blocks around it. Is it a recommended way?

Reply:
Moving up in the tree in recursive manner will throw an exception always at the end node (root)